### PR TITLE
#1466 - Pinpoint Lambda Speedup

### PR DESCRIPTION
**Pull request by EvanParish**
# Description

The sqs resource was being defined in the lambda's handler function and therefore was reinstantiated every time the lambda would run. This moved the sqs resource to the execution environment so the lambda will run more efficiently when it's not running from a cold start.

issue #1466 

## How Has This Been Tested?

This doesn't change any logic, just moves the sqs resource initialization outside of the lambda handler.
- [x] [deployed to dev](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/11110405006)

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py [hunk 1]
```diff
@@ -1,19 +1,51 @@
...
+    logger.setLevel('INFO')
+    logger.warning('Invalid log level specified. Defaulting to INFO.')
+
+# set up sqs resource
+queue_name = f"{os.getenv('NOTIFICATION_QUEUE_PREFIX')}{ROUTING_KEY}"
+try:
+    sqs = boto3.resource('sqs', region_name=AWS_REGION)
+    queue = sqs.get_queue_by_name(QueueName=queue_name)
+except ClientError as e:
+    print('ClientError 1:', e)
```
Comment (cris-oddball) [inline lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py:32]: should this print statement be removed?

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py [hunk 2]
```diff
@@ -1,19 +1,51 @@
...
+    logger.critical(
+        'ClientError, failed to create SQS resource or could not get sqs queue "%s". Exception: %s', queue_name, e
+    )
+    raise
+except Exception as e:
+    print('Exception 1:', e)
```
Comment (cris-oddball) [inline lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py:38]: should this print statement be removed?

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py [hunk 3]
```diff
@@ -44,7 +76,15 @@ def lambda_handler(
...
         }
+        print(envelope)
         msg = base64.b64encode(bytes(json.dumps(envelope), 'utf-8')).decode('utf-8')
-        queue.send_message(MessageBody=msg)
-
-    return {'statusCode': 200}
+        try:
+            queue.send_message(MessageBody=msg)
+        except ClientError as e:
+            print('ClientError 2:', e)
```
Comment (cris-oddball) [inline lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py:84]: should this print statement be removed?

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py [hunk 4]
```diff
@@ -44,7 +76,15 @@ def lambda_handler(
...
+            logger.critical('ClientError, failed to send message to SQS queue "%s". Exception: %s', queue_name, e)
+            raise
+        except Exception as e:
+            print('Exception 2:', e)
```
Comment (cris-oddball) [inline lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py:88]: should this print statement be removed?

**Review by cris-oddball — APPROVED** [review]

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py (hunk 1 above)
Comment (EvanParish) [reply lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py]: removed

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py (hunk 2 above)
Comment (EvanParish) [reply lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py]: removed

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py (hunk 3 above)
Comment (EvanParish) [reply lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py]: removed

File: lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py (hunk 4 above)
Comment (EvanParish) [reply lambda_functions/pinpoint_callback/pinpoint_callback_lambda.py]: removed

**Review by kalbfled — APPROVED** [review]

**Pull request merged**